hudi-client/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -253,6 +254,15 @@ public SyncableFileSystemView getHoodieView() {
     return getViewManager().getFileSystemView(metaClient);
   }
 
+  private SyncableFileSystemView getFileSystemViewInternal(HoodieTimeline timeline) {
+    if (config.useFileListingMetadata()) {
+      FileSystemViewStorageConfig viewConfig = config.getViewStorageConfig();
+      return new HoodieMetadataFileSystemView(metaClient, timeline, viewConfig.isIncrementalTimelineSyncEnabled());
+    } else {
+      return getViewManager().getFileSystemView(metaClient);
```

Comment:
Can we make the viewManager hand the metadataFSView as well

Reply:
Done.

Reply:
Not currently as entire metadata code is within hudi-client and viewManager is within hudi-common.

I will have to eventually restructure into metadata-reader and metadata-writer and I will make this change then.